tools/codegen/core/gen_grpc_tls_credentials_options.py

```diff
@@ -242,6 +242,19 @@ class DataMember:
         test_value_1="false",
         test_value_2="true",
     ),
+    DataMember(
+        name="sni_override",
+        type="std::optional<std::string>",
+        setter_move_semantics=True,
+        default_initializer="std::nullopt",
```

Comment:
I don't think we need a default initializer, since nullopt is the default for `std::optional<>`.